internal/network/manifests_generator.go

```diff
@@ -142,10 +147,14 @@ spec:
             [Unit]
             Description=Run dnsmasq to provide local dns for Single Node OpenShift
             Before=kubelet.service crio.service
-            After=network.target
+            After=network.target nodeip-configuration.service
 
             [Service]
+            TimeoutStartSec=30
+            ExecStartPre=/bin/bash -c 'until [ -e /var/run/nodeip-configuration/primary-ip ]; do sleep 1; done'
+            ExecStartPre=/bin/bash -c 'cp /etc/default/single-node.conf_template /etc/dnsmasq.d/single-node.conf && sed -i "s/HOST_IP/$(cat /var/run/nodeip-configuration/primary-ip)/g" /etc/dnsmasq.d/single-node.conf'
```

Comment:
```suggestion
            ExecStartPre=/bin/bash -c 'sed -i "s/HOST_IP/$(cat /var/run/nodeip-configuration/primary-ip)/g" /etc/default/single-node.conf_template > /etc/dnsmasq.d/single-node.conf'
```

Just incase sed fails, it's better we have an empty file rather than a file with an unprocessed template

Reply:
sounds great, changing